### DISCUSSION TITLE
Users should be able to add `InputObject` to `ActiveModel`

### DISCUSSION THREAD

**Discussion by JonasCir** [SeaQL/sea-orm#1356] [Ideas]
Hey there,
I have I side project where I try out different frameworks, currently it runs on `sea-orm` and `async-graphql`. 

Thanks to @tsar-boomba 's work for #1120, I can easily derive `SimpleObject` for my `Model` while generating entities.
@tqwewe outlined an approach how `ActiveModel` could derive `InputObject` in #87, but this would require code changes in `sea-orm` itself.

It would be great if a mechanism similar to #1120 would exists for `ActiveModel`. On a somewhat related note, I noticed that the generation using the expanded format does not output an `ActiveModel` struct, although it is listed in the [docs](https://www.sea-ql.org/SeaORM/docs/generate-entity/expanded-entity-structure/). This inconsistency currently prevents me from adding the `InputObject` derive myself.

TL;DR;

1. Should we introduce something like #1120 for `ActiveModel`?
2. Should `ActiveModel` be included in the expanded entity structure such that people can add their own derives there?

**Comment by billy1624** [comment]
Hey @JonasCir, I strongly recommend you to check https://github.com/SeaQL/seaography, a async-graphql extension for SeaORM.

**Reply by JonasCir** [reply]
Thank you @billy1624 for your answer. I have used `seaography` in the past, but it doesn't really fits my use case as I need full control over the resolvers. Sorry for the discussion probably being too much focused on graphql (this is my main motivation though). I still think the two points I raised are valid for other derives? I thing that devs might want to roll their own, special validation logic etc. Therefore, not being able to access the `ActiveModel` in any form really blocks them.
